Implement Clock.startTime to get the application's start time

mathias-lang-sociomantic opened the pull request:
Replace a common requirement between old and new logger, using a more appropriate module.

I started to remove old loggers from v3.x.x and it turned out `LogEvent` still had a dependency.
Moving everything to `Clock` seemed like the best compromise.

mathias-lang-sociomantic committed:
Implement Clock.startTime to get the application's start time

Replace a common requirement between old and new logger, using a more appropriate module.

nemanja-boric-sociomantic commented on src/ocean/time/Clock.d:
```diff
@@ -39,6 +39,20 @@ import ocean.core.Exception_tango;
 struct Clock
 {
+    /// Time at which the program started
+    private static Time start_time_;
```
Indentation level seems not to match the existing one.

mathias-lang-sociomantic replied:
Ah yes the old Tango modules with their own indentation. Fixed.

nemanja-boric-sociomantic replied:
Thanks!

mathias-lang-sociomantic committed:
Remove useless debug block in Clock

nemanja-boric-sociomantic commented on src/ocean/util/log/Event.d:
```diff
@@ -40,7 +40,7 @@ public struct LogEvent
     /// Set the various attributes of this event.
     void set (ILogger.Context host, ILogger.Level level, cstring msg, cstring name)
     {
-        time_ = Log.time;
```
Could we deprecate `Log.time` as well?

mathias-lang-sociomantic replied:
I plan to deprecate the whole `Log` soon.

mathias-lang-sociomantic replied:
Well actually it cannot hurt, will do

nemanja-boric-sociomantic commented:
Also :police_car: is behind you

mathias-lang-sociomantic commented:
> Also :police_car: is behind you

Why ?

nemanja-boric-sociomantic commented:
Ha, no, it seems they are there.

nemanja-boric-sociomantic commented:
Yeah, sorry, I've somehow missed them!

nemanja-boric-sociomantic commented on src/ocean/util/log/Log.d:
```diff
@@ -334,6 +328,7 @@ public struct Log
 
         ***********************************************************************/
 
+        deprecated("Use ocean.time.Clock : Clock.startTime() directly")
```
`Clock.now` is the replacement.

nemanja-boric-sociomantic replied:
Ping

mathias-lang-sociomantic replied:
Oops, fixed

mihails-strasuns-sociomantic commented:
`object.Exception: Cyclic dependency in module ocean.time.WallClock for import ocean.time.Clock` is it related to the PR or just old issue inconvered?

mathias-lang-sociomantic committed:
DateTime_tango: Replace import of WallClock with core one

The issue with the WallClock import is that it tends to introduce a cyclic dependency,
and since it has a module ctor this is a no go.
Since DateTime_tango is only used one single, convenience function, the replacement is trivial.

mathias-lang-sociomantic commented:
> is it related to the PR or just old issue inconvered?

Is "both" a valid answer ? In any case, I fixed it (see first commit, which obviously Github lists last...).

nemanja-boric-sociomantic commented on relnotes/logevent-started.deprecation.md:
```diff
@@ -0,0 +1,7 @@
+* `ocean.util.log.Log : Log.time`
+
+  This function have been deprecated. Use `ocean.time.Clock : Clock.startTime` instead.
```
Ah, here as well.

mathias-lang-sociomantic replied:
At least I was consistent. Fixed

mathias-lang-sociomantic commented:
Ping

nemanja-boric-sociomantic commented:
LGTM.
